pkg/util/instancemetadata/prod.go

```diff
@@ -71,7 +72,7 @@ func (p *prod) populateTenantIDFromMSI(ctx context.Context) error {
 	c := &azureclaim.AzureClaim{}
 	_, _, err = parser.ParseUnverified(token.OAuthToken(), c)
 	if err != nil {
-		return err
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalToken, "properties.servicePrincipalProfile", err.Error())
```

Comment:
This function is called by the RP / Gateway /Monitor, etc. when they are first starting up and performing MSI authentication so there is no concept of an API or end user. This should stay as a regular Internal Server Error.

Reply:
I changed it back to returning a regular error.